tools/querytee/response_comparator.go

```diff
@@ -0,0 +1,208 @@
+package querytee
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/pkg/labels"
+
+	"github.com/cortexproject/cortex/pkg/util"
+)
+
+// ResponsesComparatorFunc helps with comparing different responses from various routes.
+type ResponsesComparatorFunc func(expected, actual []byte) error
+
+// SamplesComparatorFunc helps with comparing different types of samples coming from /api/v1/query and /api/v1/query_range routes.
+type SamplesComparatorFunc func(expected, actual json.RawMessage) error
+
+type SamplesResponse struct {
+	Status string
+	Data   struct {
+		ResultType string
+		Result     json.RawMessage
+	}
+}
+
+var samplesComparator = map[string]SamplesComparatorFunc{
```

Comment:
Can we avoid using package-scoped global variables, and introduce a type for this map? Both `RegisterSamplesComparator` and `CompareSamplesResponse` will be methods on the type.